Corona-Warn-App/src/main/java/de/rki/coronawarnapp/coronatest/execution/TestResultScheduler.kt

```diff
@@ -2,57 +2,86 @@ package de.rki.coronawarnapp.coronatest.execution
-import de.rki.coronawarnapp.worker.DiagnosisTestResultRetrievalPeriodicWorker
 import kotlinx.coroutines.runBlocking
 import timber.log.Timber
 import java.util.concurrent.TimeUnit
 import javax.inject.Inject
 
 @Reusable
 class TestResultScheduler @Inject constructor(
-    private val tracingSettings: TracingSettings,
     private val workManager: WorkManager,
-    private val timeStamper: TimeStamper,
 ) {
 
-    private suspend fun isScheduled(): Boolean {
-        val workerInfos = workManager.getWorkInfosForUniqueWork(PCR_TESTRESULT_WORKER_UNIQUEUNAME).await()
+    private var ratWorkerMode = DISABLED
+    val ratResultPeriodicPollingMode
+        get() = ratWorkerMode
 
-        return workerInfos.any { it.isScheduled }
-    }
+    private suspend fun isPcrScheduled() =
+        workManager.getWorkInfosForUniqueWork(PCR_TESTRESULT_WORKER_UNIQUEUNAME)
+            .await()
+            .any { it.isScheduled }
+
+    private val WorkInfo.isScheduled: Boolean
+        get() = state == WorkInfo.State.RUNNING || state == WorkInfo.State.ENQUEUED
 
-    fun setPeriodicTestPolling(enabled: Boolean) {
+    fun setPcrPeriodicTestPollingEnabled(enabled: Boolean) {
         if (enabled) {
             // TODO Refactor runBlocking away
-            val isScheduled = runBlocking { isScheduled() }
+            val isScheduled = runBlocking { isPcrScheduled() }
             if (isScheduled) {
                 Timber.tag(TAG).w("Already scheduled, skipping")
                 return
             }
-            Timber.tag(TAG).i("Queueing test result worker (DIAGNOSIS_TEST_RESULT_PERIODIC_WORKER)")
+            Timber.tag(TAG).i("Queueing pcr test result worker (DIAGNOSIS_TEST_RESULT_PERIODIC_WORKER)")
             workManager.enqueueUniquePeriodicWork(
                 PCR_TESTRESULT_WORKER_UNIQUEUNAME,
                 ExistingPeriodicWorkPolicy.REPLACE,
-                buildDiagnosisTestResultRetrievalPeriodicWork()
+                buildPcrTestResultRetrievalPeriodicWork()
             )
         } else {
             Timber.tag(TAG).d("cancelWorker()")
             workManager.cancelUniqueWork(PCR_TESTRESULT_WORKER_UNIQUEUNAME)
         }
     }
 
-    private fun buildDiagnosisTestResultRetrievalPeriodicWork() =
-        PeriodicWorkRequestBuilder<DiagnosisTestResultRetrievalPeriodicWorker>(
-            BackgroundWorkHelper.getDiagnosisTestResultRetrievalPeriodicWorkTimeInterval(),
+    enum class RatPollingMode {
+        DISABLED,
+        PHASE1,
+        PHASE2
+    }
+
+    fun setRatResultPeriodicPollingMode(mode: RatPollingMode) {
+        ratWorkerMode = mode
+        if (mode == DISABLED) {
+            Timber.tag(TAG).d("cancelWorker()")
+            workManager.cancelUniqueWork(RAT_RESULT_WORKER_UNIQUEUNAME)
+        } else {
+            // no check for already running workers!
+            // worker must be replaced by next phase instance
+            Timber.tag(TAG).i("Queueing rat result worker (DIAGNOSIS_TEST_RESULT_PERIODIC_WORKER)")
```

Comment:
`DIAGNOSIS_TEST_RESULT_PERIODIC_WORKER` is the wrong TAG.
I think we don't need the tags at all, but this is not something we need to change in this PR.

Reply:
done!